plugin/platforms/android/java/com/nativescript/documentpicker/FilePath.java

```diff
@@ -282,33 +283,40 @@ else if (isDownloadsDocument(uri)) {
             }
             // MediaProvider
             else if (isMediaDocument(uri)) {
+
                 final String docId = DocumentsContract.getDocumentId(uri);
                 final String[] split = docId.split(":");
                 final String type = split[0];
 
+                Log.i(TAG, type);
                 Uri contentUri = null;
                 if ("image".equals(type)) {
                     contentUri = MediaStore.Images.Media.EXTERNAL_CONTENT_URI;
                 } else if ("video".equals(type)) {
                     contentUri = MediaStore.Video.Media.EXTERNAL_CONTENT_URI;
                 } else if ("audio".equals(type)) {
                     contentUri = MediaStore.Audio.Media.EXTERNAL_CONTENT_URI;
+                } else {
+                    contentUri = MediaStore.Files.getContentUri("external");
                 }
 
                 final String selection = "_id=?";
                 final String[] selectionArgs = new String[] {
                         split[1]
                 };
 
-                return getDataColumn(context, contentUri, selection, selectionArgs);
+                try {
+                    return getDataColumn(context, contentUri, selection, selectionArgs);
+                } catch (Exception e) {
+                    return getMediaStore(context, contentUri, null, null);
```

Comment:
would nt it be better to test for android 11 instead of waiting for the error to use `getMediaStore`?

Reply:
This bug doesn't appear in all android 11 devices. When I had been testing with an emulator on mac, works fine. When I had been testing on Pixel device fail. For this reason, I prefer to handle the exception.

Reply:
ok good with me